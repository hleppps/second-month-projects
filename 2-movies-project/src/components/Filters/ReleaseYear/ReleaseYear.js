import React from "react";
import { Form } from "react-bootstrap";

const ReleaseYear = React.memo((props) => {
  const onChangeHandler = (event) => {
    props.setYearFilter(event.target.value);
  };

  return (
    <Form.Group className="my-3">
      <Form.Label>Release year:</Form.Label>
      <Form.Select
        data-testid="year-select"
        aria-label="Default select example"
        onChange={onChangeHandler}
        value={props.filteredYear}
      >
        <option value="2021">2021</option>
        <option value="2020">2020</option>
        <option value="2019">2019</option>
        <option value="2018">2018</option>
      </Form.Select>
    </Form.Group>
  );
});
export default ReleaseYear;
