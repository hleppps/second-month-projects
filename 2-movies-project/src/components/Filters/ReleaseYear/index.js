import ReleaseYear from "./ReleaseYear";

export default ReleaseYear;
