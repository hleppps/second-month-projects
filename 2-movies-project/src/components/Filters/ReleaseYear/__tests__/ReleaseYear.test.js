import "@testing-library/jest-dom";
import { cleanup, fireEvent, render } from "@testing-library/react";
import ReleaseYear from "../ReleaseYear";

afterEach(cleanup);

describe("ReleaseYear component", () => {
  it("renders year value from props", () => {
    const year = "2019";

    const { getByTestId } = render(<ReleaseYear filteredYear={year} />);

    expect(getByTestId("year-select")).toHaveValue(year);
  });

  it("calls the year change function on change", () => {
    const yearChangeHandler = jest.fn();

    const { getByTestId } = render(
      <ReleaseYear setYearFilter={yearChangeHandler} />
    );

    fireEvent.change(getByTestId("year-select"));

    expect(yearChangeHandler).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<ReleaseYear />);
    expect(asFragment(<ReleaseYear />)).toMatchSnapshot();
  });
});
