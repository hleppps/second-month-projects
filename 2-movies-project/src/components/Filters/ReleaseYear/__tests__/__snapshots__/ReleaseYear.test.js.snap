// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseYear component matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3"
  >
    <label
      class="form-label"
    >
      Release year:
    </label>
    <select
      aria-label="Default select example"
      class="form-select"
      data-testid="year-select"
    >
      <option
        value="2021"
      >
        2021
      </option>
      <option
        value="2020"
      >
        2020
      </option>
      <option
        value="2019"
      >
        2019
      </option>
      <option
        value="2018"
      >
        2018
      </option>
    </select>
  </div>
</DocumentFragment>
`;
